src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -3480,5 +3486,745 @@ void F()
+@"#pragma warning disable 618
+using System;
+using System.Security.Permissions;
+class Program
+{
+    static void Main()
+    {
+        Action a1 = [PermissionSet(SecurityAction.Deny)] () => { };
+    }
+}";
+            var comp = CreateCompilationWithMscorlib40(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var exprs = tree.GetRoot().DescendantNodes().OfType<LambdaExpressionSyntax>().ToImmutableArray();
+            var lambda = exprs.SelectAsArray(e => GetLambdaSymbol(model, e)).Single();
+            Assert.NotEmpty(lambda.GetSecurityInformation());
+        }
+
+        [Fact]
+        public void LambdaAttributes_NullableAttributes_01()
+        {
+            var source =
+@"using System;
+using System.Diagnostics.CodeAnalysis;
+class Program
+{
+    static void Main()
+    {
+        Func<object> a1 = [return: MaybeNull][return: NotNull] () => null;
+        Func<object, object> a2 = [return: NotNullIfNotNull(""obj"")] (object obj) => obj;
+        Func<bool> a4 = [MemberNotNull(""x"")][MemberNotNullWhen(false, ""y"")][MemberNotNullWhen(true, ""z"")] () => true;
+    }
+}";
+            var comp = CreateCompilation(
+                new[] { source, MaybeNullAttributeDefinition, NotNullAttributeDefinition, NotNullIfNotNullAttributeDefinition, MemberNotNullAttributeDefinition, MemberNotNullWhenAttributeDefinition },
+                parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var exprs = tree.GetRoot().DescendantNodes().OfType<LambdaExpressionSyntax>().ToImmutableArray();
+            Assert.Equal(3, exprs.Length);
+            var lambdas = exprs.SelectAsArray(e => GetLambdaSymbol(model, e));
+            Assert.Equal(FlowAnalysisAnnotations.MaybeNull | FlowAnalysisAnnotations.NotNull, lambdas[0].ReturnTypeFlowAnalysisAnnotations);
+            Assert.Equal(new[] { "obj" }, lambdas[1].ReturnNotNullIfParameterNotNull);
+            Assert.Equal(new[] { "x" }, lambdas[2].NotNullMembers);
+            Assert.Equal(new[] { "y" }, lambdas[2].NotNullWhenFalseMembers);
+            Assert.Equal(new[] { "z" }, lambdas[2].NotNullWhenTrueMembers);
+        }
+
+        [Fact]
+        public void LambdaAttributes_NullableAttributes_02()
+        {
+            var source =
+@"#nullable enable
+using System;
+using System.Diagnostics.CodeAnalysis;
+class Program
+{
+    static void Main()
+    {
+        Func<object> a1 = [return: MaybeNull] () => null;
+        Func<object?> a2 = [return: NotNull] () => null;
+    }
+}";
+            var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition, NotNullAttributeDefinition }, parseOptions: TestOptions.RegularPreview);
+            // PROTOTYPE: Report error for a2, no error for a1.
```

Comment:
I think both declarations will end up having a warning. The first one in the conversion/assignment, and the other in the return value of the lambda.